articles/architecture-scenarios/application/web-app-sso.md

```diff
@@ -0,0 +1,410 @@
+:::
+
+All users will implicitly be regular users, but timesheet administrators will be assigned to an `Admin` group which will allow them to approve timesheets. The Authorization Extension allows for mapping groups to existing group membership.
+
+All timesheet administrators will be assigned to the `Timesheet Administrators` group on Active Directory, which will be automatically mapped to the `Admin` group inside the Timesheet Application.
+
+When you install the Authorization Extension, it creates a rule in the background, which does the following:
+1. Determine the user's group membership.
+1. Store the user's group membership info as part of the `app_metadata`.
+1. Add the user's group membership to the outgoing token.
+1. Verify that the user has been granted access to the current application.
+
+
+#### Refresh the authorization related claims
+
+In the scenario described in this document, the web app does not handle its own authorization, but instead Auth0 delivers the authorization claims. User group membership is retrieved from the Active Directory and passed in the web app as claims in the `id_token`.
+
+There are some scenarios however when you might need to refresh the user's authorization claims. For example:
+- The user leaves the company so all access should be revoked.
+- The user receives Admin privileges and you want this change to be propagated quickly.
+- You consider the timesheets approval process a sensitive one, so each time a user tries to approve an entry, you want to verify that the user indeed has the admin rights required.
+
+In all these scenarios, the user is already logged in so the app has parsed and applied the access control granted by the claims contained in the `id_token`, but you want to refresh these access control related claims.
+
+In this case the developer can pick one of the following solution to this problem:
+
+- The web app falls back on the Auth0 SSO session when it wants to refresh the `id_token` by performing another authorization flow (`/authorize` endpoint) with Auth0.
+
+- When performing the initial authorization flow, the web app requests a `refresh_token`, by adding `offline_access` at the `scope` parameter, for example `scope=openid offline_access`. The `refresh_token` is stored in session, alongside with the `id_token`. Then when a session needs to be refreshed (a timeframe has passed or the user tries to approve a timesheet), the app uses the `refresh_token` on the backend to obtain a new `id_token`, using the `/oauth/token` endpoint with `grant_type=refresh_token`. This will result in re-querying the Active Directory, get the updated information, which would include any new groups the user has been added to since the last time, and sending this information to the web app as part of the `id_token` claims.
```

Comment:
This needs to be tested against pipeline2 https://auth0.slack.com/archives/patterns-practices/p1475506975000342